flowetl/etl/tests/conftest.py

```diff
@@ -1,14 +1,73 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+conftest for unit tests
+"""
 import pytest
 
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+from etl.model import Base, ETLRecord  # pylint: disable=unused-import
 
+# pylint: disable=too-few-public-methods
 class FakeDagRun:
-    def __init__(self, conf):
+    """
+    a fake DagRun object used for faking dag config
+    """
+
+    def __init__(self, conf=None):
         self.conf = conf
 
 
+class FakeTaskInstance:
+    """
+    a fake TaskInstance object
+    """
+
+    def __init__(self, task_id):
+        self.task_id = task_id
+
+
 @pytest.fixture(scope="function")
 def create_fake_dag_run():
-    def fake_dag_run(*, conf):
+    """
+    Returns a function that can be used to generate a fake DagRun with
+    specific config
+    """
+
+    def fake_dag_run(*, conf=None):
         return FakeDagRun(conf=conf)
```

Comment:
I'm not advocating anything here, but any special reason this doesn't just return `Mock(conf=conf)`?

Reply:
It did but max pointed out that this was overkill for something so simple...

Reply:
I suggested removing the mock in a previous PR (can't find the link right now), although this was primarily because the way in which the attribute was set on the mock was a bit roundabout and make the code hard to read.

I had no idea you can just pass values for attributes which you want set during initialisation of the mock. 😮 That makes it much more readable, so I'd be fine with that solution too. 👌 

Another reason (although I'll readily admit a much more pernickety one) is that technically this isn't a mock (see my comment [here](https://github.com/Flowminder/FlowKit/pull/774#discussion_r283762082)). But I agree for pragmatic and convenience reasons it would be totally fine to use a `Mock` here if we want to.

Reply:
I am just going to leave it as it is...both do the same thing...It's not difficult to understand what is going on.

Reply:
Could also use `types.SimpleNamespace(conf=conf)`?

Reply:
I could but that's less understandable...